app/scripts/modules/core/src/bootstrap/uiSelect.config.ts

```diff
@@ -20,6 +21,17 @@ function uiSelectDecorator($provide: ng.auto.IProvideService) {
           }
           if (!event.defaultPrevented) {
             $select.selected.push(item);
+            const locals = {
+              [$select.parserResult.itemName]: item,
+            };
+            $timeout(function() {
+              if ($select.onSelectCallback) {
+                $select.onSelectCallback(scope, {
+                  $item: item,
+                  $model: $select.parserResult.modelMapper(scope, locals),
+                });
+              }
+            });
```

Comment:
This change is the only bits missing from the [original default implementation](https://github.com/angular-ui/ui-select/blob/master/src/uiSelectMultipleDirective.js#L170-L185) in `uiSelect` and is responsible for invoking the `on-select` callback.